src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,27 +8,61 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
+    public static final String EXAMPLE = "123, some street, #00-00, 000000";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    
+    private static final int BLOCK_INDEX = 0;
+    private static final int STREET_INDEX = 1;
+    private static final int UNIT_INDEX = 2;
+    private static final int POSTAL_CODE_INDEX = 3;
 
-    public final String value;
+    private final Block block_;
+    private final Unit unit_;
+    private final Street street_;
+    private final PostalCode postalCode_;
+    
+    private final String value_;
+    
     private boolean isPrivate;
 
     /**
-     * Validates given address.
+     * Validates given address. e.g. a/BLOCK, STREET, UNIT, POSTAL_CODE
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
+        String[] splittedAddress = address.split(",");
+        value_ = address.trim();
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        if (!isValidAddress(value_)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        this.block_ = new Block(splittedAddress[BLOCK_INDEX].trim());
+        this.street_ = new Street(splittedAddress[STREET_INDEX].trim());
+        this.unit_ = new Unit(splittedAddress[UNIT_INDEX].trim());
+        this.postalCode_ = new PostalCode(splittedAddress[POSTAL_CODE_INDEX].trim());
     }
 
+    public Block getBlock(){
```

Comment:
Good job providing accessors for individual parts of the address. Many in the class have not done this.